engine/src/main/java/io/camunda/zeebe/engine/state/migration/DbMigrationState.java

```diff
@@ -294,4 +324,18 @@ public void migrateDrgPopulateDrgVersionByDrgIdAndKey() {
               decisionRequirementsIdAndVersion, fkDecisionRequirements);
         });
   }
+
+  @Override
+  public void migrateElementInstancePopulateProcessInstanceByDefinitionKey() {
+    elementInstanceColumnFamily.forEach(
+        (key, record) -> {
+          final var recordValue = record.getValue();
+          if (recordValue.getBpmnElementType() == BpmnElementType.PROCESS) {
+            elementInstanceKey.wrapLong(key.getValue());
+            processDefinitionKey.wrapLong(recordValue.getProcessDefinitionKey());
+            processInstanceKeyByProcessDefinitionKeyColumnFamily.insert(
+                processInstanceKeyByProcessDefinitionKey, DbNil.INSTANCE);
+          }
+        });
```

Comment:
> would be willing to share his thoughts on how large our states can be nowadays.

That totally depends on the workload, so not sure how to answer this :)

Reply:
@Zelldon Perhaps you can share the maximum you encountered in your recent tests. It may be a ballpark number for us to use in testing how long migration of such a large state would take.

Reply:
Still, this question is not clear to me. It can be as large as we want to go 😅 

In one of my tests for example I had a 80 gig snapshot. Filled with 3 partitions a 500 gig disk. 

In previous versions, we had issues with 1 gig of snapshot, due to the performance bottleneck for RocksDB iterations, which should now be solved. 

I can't answer what a good size is since this totally depends on the use case and the customer. Might make sense to check in SaaS what the avg snapshot size is. Or you just go with 5 to 10 gig 🤷🏼 but then this is just a guess again. I would say at least go over the 1 gig.

Reply:
Thanks for this discussion. I will leave the PR open for now and run some tests once I've progressed further along with the deletion topic . Like @korthout I'm also not sure how I could improve this further currently.